### PR TITLE
Do not delete module resource until all managed resources are gone

### PR DESCRIPTION
**Pull request by hisarbalik**
**Description**

Changes proposed in this pull request:

- Telemetry manager will check all managed resources before deleting telemetry resource
- When any managed resource still in use or has an instance, telemetry manager will update status to 'Error'

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/17162

### CODE REVIEW

File: test/e2e/telemetry_test.go [hunk 1]
```diff
@@ -100,13 +103,78 @@ var _ = Describe("Telemetry-module", func() {
...
+			Eventually(func(g Gomega) {
+				var telemetry v1alpha1.Telemetry
+				g.Expect(k8sClient.Get(ctx, telemetryKey, &telemetry)).Should(Succeed())
+				g.Expect(telemetry.Status.State).Should(Equal(v1alpha1.StateError))
+				g.Expect(telemetry.Finalizers).Should(HaveLen(1))
+				g.Expect(telemetry.Finalizers[0]).Should(Equal("telemetry.kyma-project.io/finalizer"))
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("Should delete telemetry", func() {
```
Comment (skhalash) [inline test/e2e/telemetry_test.go:162]: Let's capitalize Telemetry in test case description to stress out that the CR is meant

File: internal/reconciler/telemetry/reconciler.go [hunk 2]
```diff
@@ -210,3 +218,72 @@ func (r *Reconciler) serverSideApply(ctx context.Context, obj client.Object) err
...
+	if err := r.List(ctx, &tracePipelineList); err != nil {
+		//no kind found
+		if _, ok := err.(*meta.NoKindMatchError); ok {
+			return false
+		}
+	}
+
+	if len(tracePipelineList.Items) > 0 {
+		return true
+	}
```
Comment (skhalash) [inline internal/reconciler/telemetry/reconciler.go:286]:
```suggestion
	return len(tracePipelineList.Items) > 0
```

File: internal/reconciler/telemetry/reconciler.go [hunk 3]
```diff
@@ -210,3 +218,72 @@ func (r *Reconciler) serverSideApply(ctx context.Context, obj client.Object) err
 	obj.SetResourceVersion("")
 	return r.Patch(ctx, obj, client.Apply, client.ForceOwnership, client.FieldOwner(fieldOwner))
 }
+
+func (r *Reconciler) checkAnyResourceExist(ctx context.Context) bool {
+	return r.checkLogParserExist(ctx) ||
+		r.checkLogPipelineExist(ctx) ||
+		r.checkMetricPipelinesExist(ctx) ||
+		r.checkTracePipelinesExist(ctx)
+}
```
Comment (skhalash) [inline internal/reconciler/telemetry/reconciler.go:227]: Let's fix the wording here - either CustomResourcesExist or CustomResourceExists

**Review by skhalash — CHANGES_REQUESTED** [review]

File: test/e2e/telemetry_test.go (hunk 1 above)
Comment (chrkl) [reply test/e2e/telemetry_test.go]: Done

**Pull request merged**